tools/snippet-testing/snippet_tester.rb

```diff
@@ -61,6 +61,16 @@ def test_folder(snippet)
     end
   end
 
+  def add_hosts()
+    Dir.chdir(__dir__) do
+      `sudo -- sh -c "cat ../hosts >> /etc/hosts"`
+    end
+  end
+
+  def remove_hosts()
+    `sudo -- sh -c "sed -i '' '/^# START: api-snippets$/,/^# END: api-snippets$/d' /etc/hosts"`
```

Comment:
Thanks for your observation. Actually, there is an error regarding mac `sed` and linux `sed` both are incompatible when using `-i` flag.
I already worked on this and I'll keep the linux version, and updated the readme to include a solution for mac users.
Regarding the command, you can see it like

```
sed
  -i'' # option and argument, notice there's no space! that's how it works in linux
  '/regex/,/regex/d' # regex range
  pathfile # the file sed will read
```